lib/buildingsync/makers/workflow_maker_phase_zero.rb

```diff
@@ -0,0 +1,906 @@
+#  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+#  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+########################################################################################################################
+require_relative '../workflow_maker'
+module BuildingSync
+  # base class for objects that will configure workflows based on building sync files
+  class PhaseZeroWorkflowMaker < WorkflowMaker
+    def initialize(doc, ns)
+      super
+
+      # load the workflow
+      @workflow = nil
+
+      # log failed scenarios
+      @failed_scenarios = []
+
+      # select base osw for standalone, small office, medium office
+      base_osw = 'phase_zero_base.osw'
+
+      workflow_path = File.join(File.dirname(__FILE__), base_osw)
+      raise "File '#{workflow_path}' does not exist" unless File.exist?(workflow_path)
+
+      File.open(workflow_path, 'r') do |file|
+        @workflow = JSON.parse(file.read)
+      end
+
+      @facility = {}
+      @subsections = []
+
+      # configure the workflow based on properties in the xml
+      configureForDoc(@workflow)
+    end
+
+    def configureForDoc(osw)
+
+      # get the facility floor areas
+      @facility['gross_floor_area'] = nil
+      @facility['heated_and_cooled_floor_area'] = nil
+      @facility['footprint_floor_area'] = nil
+      @doc.elements.each("/#{@ns}:BuildingSync/#{@ns}:Facilities/#{@ns}:Facility/#{@ns}:Sites/#{@ns}:Site/#{@ns}:Buildings/#{@ns}:Building/#{@ns}:FloorAreas/#{@ns}:FloorArea") do |floor_area_element|
+        floor_area = floor_area_element.elements["#{@ns}:FloorAreaValue"].text.to_f
+        next if floor_area.nil?
+
+        floor_area_type = floor_area_element.elements["#{@ns}:FloorAreaType"].text
+        if floor_area_type == 'Gross'
+          @facility['gross_floor_area'] = floor_area
+        elsif floor_area_type == 'Heated and Cooled'
+          @facility['heated_and_cooled_floor_area'] = floor_area
+        elsif floor_area_type == 'Footprint'
+          @facility['footprint_floor_area'] = floor_area
+        end
+      end
+
+      # SHL- get the template (vintage)
+      @facility['template'] = nil
+      @doc.elements.each("/#{@ns}:BuildingSync/#{@ns}:Facilities/#{@ns}:Facility/#{@ns}:Sites/#{@ns}:Site/#{@ns}:Buildings/#{@ns}:Building") do |facility_element|
+        built_year = facility_element.elements["#{@ns}:YearOfConstruction"].text.to_f
+
+        if facility_element.elements["#{@ns}:YearOfLastMajorRemodel"]
+          major_remodel_year = facility_element.elements["#{@ns}:YearOfLastMajorRemodel"].text.to_f
+          built_year = major_remodel_year if major_remodel_year > built_year
+        end
+
+        if built_year < 1978
```

Comment:
this appears to be very California centric. We need to make sure we abstract out this logic for ASHRAE climate zones.

Reply:
Agree

Reply:
see #9